slatedb/src/utils.rs

```diff
@@ -95,9 +95,18 @@ where
     //       details.
     let wrapped = AssertUnwindSafe(future).catch_unwind().map(move |outcome| {
         let result = match outcome {
-            Ok(Ok(val)) => Ok(val),
-            Ok(Err(e)) => Err(e),
-            Err(panic) => Err(BackgroundTaskPanic(name, Arc::new(Mutex::new(panic)))),
+            Ok(result) => result,
+            Err(payload) => {
+                if let Some(err) = payload.downcast_ref::<SlateDBError>() {
+                    return Err(err.clone());
+                }
+                error!(
```

Comment:
I opted to log the panic here. This function is only used in compactor executor. We treat the executor failures as non-fatal (in compactor.rs). So, I simply return the task panic error with the task name for now. Since this bg task is not really part of the dispatcher pool (it's a short lived, spawned task for executors) it's fundamentally different from the long-running DB BG tasks (like compaction, gc, etc).